components/embedder_traits/lib.rs

```diff
@@ -154,6 +154,9 @@ pub enum EmbedderMsg {
     GetSelectedBluetoothDevice(Vec<String>, IpcSender<Option<String>>),
     /// Open file dialog to select files. Set boolean flag to true allows to select multiple files.
     SelectFiles(Vec<FilterPattern>, bool, IpcSender<Option<Vec<String>>>),
+    /// Open yes/no message for user to allow permission specified by first String.
+    /// With dialog title specified by second String.
+    PromptPermission(String, String, IpcSender<PermissionRequest>),
```

Comment:
I propose instead of sending two strings, we send an enum:
```rust
enum PermissionName {
   // variants that match the DOM PermissionName enum
}

enum PermissionPrompt {
    Insecure(PermissionName),
    Request(PermissionName),
}
```

This gives embedders the information they require in order to display whatever native prompt interface would make sense on that platform and supports localization better in the future.